src/OBO.NET/OboOntology.fs

```diff
@@ -62,7 +200,12 @@ type OboOntology =
 
         let terms, typedefs = loop [] [] entries
 
-        OboOntology.create terms typedefs
+        OboOntology.create terms typedefs formatVersion dataVersion ontology (date : DateTime option) (savedBy : string option) (autoGeneratedBy : string option) subsetdefs imports synonymtypedefs idSpaces (defaultRelationshipIdPrefix : string option) idMappings remarks treatXrefsAsEquivalents treatXrefsAsGenusDifferentias treatXrefsAsRelationships treatXrefsAsIsAs relaxUniqueIdentifierAssumptionForNamespaces relaxUniqueLabelAssumptionForNamespaces
+
+    /// Takes a list of OboEntries and the document header tags and returns the OboOntology based on them.
+    static member FromOboEntries(entries, formatVersion, ?DataVersion, ?Ontology, ?Date, ?SavedBy, ?AutoGeneratedBy, ?Subsetdefs, ?Imports, ?Synonymtypedefs, ?Idspaces, ?DefaultRelationshipIdPrefix, ?IdMappings, ?Remarks, ?TreatXrefsAsEquivalents, ?TreatXrefsAsGenusDifferentias, ?TreatXrefsAsRelationships, ?TreatXrefsAsIsAs, ?RelaxUniqueIdentifierAssumptionForNamespaces, ?RelaxUniqueLabelAssumptionForNamespaces) =
+        //let subsetdefs = match Subsetdefs with None -> [] | _ -> Subsetdefs.Value
+        OboOntology.fromOboEntries formatVersion DataVersion Ontology Date SavedBy AutoGeneratedBy (defaultArg Subsetdefs []) (defaultArg Imports []) (defaultArg Synonymtypedefs []) (defaultArg Idspaces []) DefaultRelationshipIdPrefix (defaultArg IdMappings []) (defaultArg Remarks []) (defaultArg TreatXrefsAsEquivalents []) (defaultArg TreatXrefsAsGenusDifferentias []) (defaultArg TreatXrefsAsRelationships []) (defaultArg TreatXrefsAsIsAs []) (defaultArg RelaxUniqueIdentifierAssumptionForNamespaces []) (defaultArg RelaxUniqueLabelAssumptionForNamespaces []) entries
 
     /// Writes an OBO Ontology to term and type def stanzas in line form.
     static member toLines (oboOntology : OboOntology) =

```

Comment:
this should also reflect the new header additions, otherwise roundtrip tests will fail